python/file_finder.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+import os
+import subprocess
+import sys
+
+#From: https://stackoverflow.com/questions/1883980/find-the-nth-occurrence-of-substring-in-a-string
+def find_nth(haystack, needle, n):
+    start = haystack.find(needle)
+    while start >= 0 and n > 1:
+        start = haystack.find(needle, start+len(needle))
+        n -= 1
+    return start
+
+def find_site(file_per_job, prefer_us_sites = False):
+    file_and_site_per_job = {}
+    print "Finding the sites for each file ...",
+    sys.stdout.flush()
+    for i, (job, file) in enumerate(file_per_job.iteritems()):
+        if file is None:
+            file_and_site_per_job[job] = (file,None,[None])
+        else:
+            cmd = "dasgoclient -query=\"site file=" + file + "\""
+            p = subprocess.Popen(cmd, shell = True, stdout=subprocess.PIPE)
+            out, err = p.communicate()
+            sites = [None] if "WARNING:" in out else out.split()
+            site = select_site(sites, prefer_us_sites)
+            file_and_site_per_job[job] = (file,site,sites)
+    print "DONE"
+    return file_and_site_per_job
+
+def get_input_file(basepath, jobs, key):
+    file_per_job = {}
+    print "Finding the input file for each job ...",
+    sys.stdout.flush()
+    for job in jobs:
+        output_file = basepath+job.stdout+".stdout"
+        if not os.path.exists(output_file):
+            file_per_job[job] = None
+        else:
+            with open(output_file, 'r') as f:
+                for line in lines_that_contain(key, f):
+                    line = line[line.find("/store/"):line.rfind(".root")+5]
+                    line = line.split(",")[0]
+                    if "/store/test/xrootd/" in line:
+                        line = line[find_nth(line,"/store/",2):]
+                    file_per_job[job] = line
+    print "DONE"
+    return file_per_job
+
+def get_input_file_from_classad(jobs, classad):
+    file_per_job = {}
+    print "Finding the input file for each job ...",
+    sys.stdout.flush()
+    for job in jobs:
+        if hasattr(job, "inputFiles"):
+            input_file = job.inputFiles.split(",")[0]
+            input_file = input_file[input_file.find("/store/"):input_file.rfind(".root")+5]
+            if "/store/test/xrootd/" in input_file:
+                input_file = input_file[find_nth(input_file,"/store/",2):]
+            file_per_job[job] = input_file
+        else:
+            file_per_job[job] = None
+    print "DONE"
+    return file_per_job        
+
+def lines_that_contain(string, fp):
+    return [line for line in fp if string in line]
+
+def select_site(sites, prefer_us_sites = False):
+    selected = None
+    sites = [s for s in sites if s is not None and "Tape" not in s]
+    sites = sorted(sites, key = lambda x: ("FNAL" in x.split('_')[2], "US" in x.split('_')[1] and prefer_us_sites), reverse = True)
```

Comment:
Just because FNAL generally has good uptime and keeps the files in good working order. Do you want me to remove this?

Reply:
I don't think it makes sense as a generic default.

Since this is now integrated in CondorProduction, you could use the ini configuration files to allow users to specify their own preferred site(s).